sources/zarr/large_image_source_zarr/__init__.py

```diff
@@ -1249,6 +1323,32 @@ def _generateDownsampledLevels(self, resample_method):
                     )
             self._validateZarr()  # refresh self._levels before continuing
 
+    def _applyGeoReferencing(self, path):
+        if self.projection is not None and self.gcps is None:
+            msg = (
+                'Projection was specified but GCPs were not specified. '
+                'Output will not be georeferenced.'
+            )
+            warnings.warn(msg, stacklevel=2)
+        if self.projection is None and self.gcps is not None:
+            msg = (
+                'GCPs were specified but projection was not specified. '
+                'Output will not be georeferenced.'
+            )
+            warnings.warn(msg, stacklevel=2)
+
+        if (
+            self.projection is not None and
+            self.gcps is not None and
+            len(self.gcps)
+        ):
+            import tifftools
+
+            tifftools.tiff_set(path, setlist=[
+                ('projection', self.projection),
+                ('gcps', self.gcps),
```

Comment:
I've used gcps without a projection.  Specifically, I have two images and I want to warp one image to match the other.  Neither is geospatial.  I can add gcps on one image and then the gdalwarp will properly reproject the image as I desire (and, eventually, I'd like to make the multisource use scikit-image do this as a tps or affine warp, too).  I'd change this to not complain if there is no projection, and to just add gcps when there is no projection specified.